src/Microsoft.Azure.SignalR.Common/RestClients/RestClientFactory.cs

```diff
@@ -0,0 +1,43 @@
+
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Net.Http;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Azure.SignalR
+{
+    public class RestClientFactory
+    {
+        private const string HttpClientName = "HttpClientWithUserAgent";
+        private IHttpClientFactory _httpClientFactory;
+        private protected readonly IHttpClientBuilder _httpClientBuilder;//for test, enable hook operations on the builder object before generating the _httpClientFactory.
+
+        internal RestClientFactory(string userAgent)
```

Comment:
is there any need to customize UserAgent? How about adding it directly into `SignalRServiceRestClient` when creating it? There is a `UserAgent` property for `ServiceClient<T>`, can we leverage that?

Reply:
`UserAgent` property of `ServiceClient<T>` isn't the same thing with `AsrsUserAgent'. The former stands for ""User-Agent" while the latter stands for "Asrs-User-Agent".

Reply:
get it. Consider adding it as part of `SignalRServiceRestClient`, it can be a mandatory header when using `SignalRServiceRestClient`